src/Sentry/Extensibility/SentryStackTraceFactory.cs

```diff
@@ -128,20 +138,35 @@ protected SentryStackFrame InternalCreateFrame(StackFrame stackFrame, bool deman
             var frame = new SentryStackFrame();
             if (GetMethod(stackFrame) is { } method)
             {
-                // TODO: SentryStackFrame.TryParse and skip frame instead of these unknown values:
                 frame.Module = method.DeclaringType?.FullName ?? unknownRequiredField;
                 frame.Package = method.DeclaringType?.Assembly.FullName;
 
-                // Include parameters in the function name
-                var parameterListFormatted = string.Join(
-                    ", ",
-                    method.GetParameters().Select(p => $"{p.ParameterType.Name} {p.Name}")
-                );
+                if (_options.StackTraceMode == StackTraceMode.Enhanced && stackFrame is EnhancedStackFrame enhancedStackFrame)
+                {
+                    var sb = new StringBuilder();
+                    frame.Function = enhancedStackFrame.MethodInfo.Append(sb, false).ToString();
+
+                    if (enhancedStackFrame.MethodInfo.DeclaringType is { } declaringType)
+                    {
+                        sb.Clear();
+                        sb.AppendTypeDisplayName(declaringType);
+                        frame.Module = sb.ToString();
```

Comment:
When I set `Module` and `Package` Sentry seems to ignore `Module` completely.

@matejminar maybe can chime in here? We're doing some changes in the .NET stack traces and it might need a bit of help on the UI to properly render.

Reply:
Sure, I'll gladly help.